builtin/logical/database/path_creds_create.go

```diff
@@ -99,6 +117,41 @@ func (b *databaseBackend) pathCredsCreateRead() framework.OperationFunc {
 	}
 }
 
+func (b *databaseBackend) pathStaticCredsRead() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		name := data.Get("name").(string)
+
+		role, err := b.StaticRole(ctx, req.Storage, name)
+		if err != nil {
+			return nil, err
+		}
+		if role == nil {
+			return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", name)), nil
+		}
+
+		dbConfig, err := b.DatabaseConfig(ctx, req.Storage, role.DBName)
+		if err != nil {
+			return nil, err
+		}
+
+		// If role name isn't in the database's allowed roles, send back a
+		// permission denied.
+		if !strutil.StrListContains(dbConfig.AllowedRoles, "*") && !strutil.StrListContainsGlob(dbConfig.AllowedRoles, name) {
+			return nil, fmt.Errorf("%q is not an allowed role", name)
```

Comment:
Should this message be a 4xx (i.e. error in first term) instead of a 500?

Reply:
I'm not sure, this is a copy-paste of [the existing `pathCredsCreateRead` error handling][1], so I assume we'd want to change both? 

[1]: https://github.com/hashicorp/vault/blob/3860fbbf05ed4b7d292dbd7de4d7e9bf96201e52/builtin/logical/database/path_creds_create.go#L53-L55

Reply:
Well, I thought so, but doing more digging I'm less sure. There is more magic going on with the responses than I'd previously thought. I'm going to make a note of this location and the other and dig into in it (this response code/object question is something I've wanted to get written down anyway). I don't think this blocks the PR especially since it's consistent.